clang/include/clang/Sema/Sema.h

```diff
@@ -2099,6 +2099,70 @@ class Sema final : public SemaBase {
   bool CheckCountedByAttrOnField(FieldDecl *FD, Expr *E, bool CountInBytes,
                                  bool OrNull);
 
+  // AssignmentAction - This is used by all the assignment diagnostic functions
+  // to represent what is actually causing the operation
+  enum AssignmentAction {
```

Comment:
@Endilll Any opinions on where `AssignmentAction` should go? Unfortunately I had to move it because the enum needs to be defined before I can use it as a parameter type.

This wasn't a problem in the original version of my patch but when I rebased on latest `next` it appears a recent refactor (9f1dc01e0ab4f91c4052a712ce590d2e86b81dc3) moved where the `BoundsSafety*` methods are declared so that they are now **before** where `AssignmentAction`'s declaration. Previously the methods were declared after `AssignmentActions`'s declaration.

Reply:
Ideally you should turn it into scoped enum and move it outside of `Sema` definition in a separate NFC patch, then update this PR with those changes.

Reply:
@Endilll I created #106453 which I think is what you're asking for.